winlogbeat/eventlog/wineventlog.go

```diff
@@ -89,10 +91,16 @@ func (l *winEventLog) Name() string {
 	return l.channelName
 }
 
-func (l *winEventLog) Open(recordNumber uint64) error {
-	bookmark, err := win.CreateBookmark(l.channelName, recordNumber)
-	if err != nil {
-		return err
+func (l *winEventLog) Open(state checkpoint.EventLogState) error {
+	var bookmark win.EvtHandle
+	var err error
+	if len(state.Bookmark) > 0 {
+		bookmark, err = win.CreateBookmarkFromXML(state.Bookmark)
```

Comment:
This is missing an error check. Looks like you intended to move the error check below outside the `if/else`.

Reply:
good catch